taichi/transforms/make_block_local.cpp

```diff
@@ -41,122 +41,142 @@ void make_block_local_offload(OffloadedStmt *offload) {
 
     // TODO: improve IR builder to make this part easier to read
 
-    // Step 1:
```

Comment:
Sorry about the large change here. The prologue generation code is moved to the `create_xlogue` lambda function with contents mostly unchanged. (This adds one more indentation level so GitHub fails to match...)

This allows us to invoke `create_xlogue` in both prologue and epilogue.